### DISCUSSION TITLE
组件命名导致的找不到组件的问题

### DISCUSSION THREAD

**Discussion by weeptao** [ant-design/ant-design#50546] [Q&A]
在nextjs14 里使用 <A.B/> 这样的组件，报错 `This is probably a bug in the React Server Components bundler.` ,像这样的组件我应该如何使用？
`const {B}=A` 也不行

**Comment by afc163** [comment]
在 Next.js 14 中，<A.B /> 这样的嵌套组件可能会因为 React Server Components 的限制或 bundler 的问题而报错，尤其是在服务端渲染中遇到这类嵌套导出。为了绕过这个问题，你可以尝试以下几种方法来解决。

1. 动态导入组件

你可以使用 dynamic 来动态导入嵌套组件。这可以确保组件只在客户端加载，避免服务端渲染时的 bundling 问题。

```jsx
import dynamic from 'next/dynamic';

const B = dynamic(() => import('path-to-A').then(mod => mod.A.B));

const MyComponent = () => (
  <div>
    <B />
  </div>
);
```

export default MyComponent;

2. 在客户端使用 <A.B />

确保组件只在客户端渲染，使用 useEffect 来避免服务端渲染时的问题。

```jsx
import { useEffect, useState } from 'react';
import A from 'path-to-A';

const MyComponent = () => {
  const [isClient, setIsClient] = useState(false);

  useEffect(() => {
    setIsClient(true); // 确保只在客户端渲染
  }, []);

  return (
    <div>
      {isClient && <A.B />}
    </div>
  );
};

export default MyComponent;
```

3. 将嵌套组件解构到顶层

由于在 Next.js 14 中可能对嵌套组件有不同的处理方式，你可以尝试手动解构组件并避免嵌套调用。像你提到的 const { B } = A 应该可以工作，但如果它不起作用，你可以尝试这样做：

```jsx
import A from 'path-to-A';

const MyComponent = () => {
  const B = A.B; // 明确解构嵌套组件

  return (
    <div>
      <B />
    </div>
  );
};

export default MyComponent;
```

4. 检查导出和模块配置

如果以上方法依然报错，检查 A 的导出方式以及项目的 module 配置。确保 A.B 组件导出正确，并且你使用的是标准的 ESM 模块导出：

```jsx
export const A = {
  B: () => <div>B Component</div>,
};
```

总结：

1.	动态导入和客户端渲染可以避免 React Server Components bundler 的问题。
2.	直接解构嵌套组件或重构导出逻辑可能会更好地适应 Next.js 14 的运行时环境。

你可以根据具体的需求和项目设置选择合适的方法。

**Comment by afc163** [comment]
https://medium.com/@fdikmen/fixing-could-not-find-the-module-error-in-nextjs-with-ant-design-3ae2cfe0160d
